### DISCUSSION TITLE
How would I best move my app linked to ionic?

### DISCUSSION THREAD

**Discussion by lostdesign** [ionic-team/capacitor#4650] [Q&A]
Hi Capacitor community <3 see a TLDR at the end if its too much text, trying to be as descriptive as possible.

I've build a macos app based on electron and wanted to bring it to mobile apps. While developing new features, I am always checking how would I best migrate over to ionic to the point where I am now, that I should move over right away.

Currently I am storing `.json` files on the client locally, the app is fully offline and opensource btw. During my research I found that there is @capacitory-community/sqlite which currently doesn't work for electron, but is in progress and web is not supported at all yet (which isn't too much of an issue for me right now).

The reason I am looking into sqlite is, that I want to add a search, which is quite performance extensive using node fs, opening the files, reading them out and checking wether it is a hit or not. Sqlite makes this a bit more easy.

But then I also found out about "secure storage" which seems to be enterprise only or is marketed as such? We also were thinking about possible syncing features as well as having everything encrypted as I don't ever want to know the users notes as well as giving them a secure option to sync their data across while never having anyone see it's contents.

I sure could just keep the electron build as is and copy over the vue part to ionic, but then I'll be maintaining two versions of the app.

tldr:
- current app is os build on electron
- files are stored as json locally
- want to support ios/android eventually
- probably sync across devices
- encrypt storage

Questions:
- Continue using json across devices?
- Use the community/sqlite feature and wait for electron support?
- Split repos and maintain two versions?
- Move away from local first and store everything in a database?

I'd appreciate any help/insight. For anyone who wants to checkout the app, you can find it here: https://github.com/muc-dev/linked

Thanks
